edocumentation-service/src/main/java/org/dcsa/edocumentation/service/BookingService.java

```diff
@@ -45,24 +43,17 @@ public Optional<BookingTO> getBooking(String carrierBookingRequestReference) {
 
   @Transactional
   public BookingRefStatusTO createBooking(BookingTO bookingRequest) {
-    OffsetDateTime now = OffsetDateTime.now();
-    Booking.BookingBuilder bookingToSave = bookingMapper.toDAO(bookingRequest).toBuilder();
+    Booking booking = bookingMapper.toDAO(bookingRequest).toBuilder()
+      .voyage(voyageService.resolveVoyage(bookingRequest))
+      .vessel(vesselService.resolveVessel(bookingRequest))
+      .placeOfIssue(locationService.ensureResolvable(bookingRequest.placeOfBLIssue()))
+      .invoicePayableAt(locationService.ensureResolvable(bookingRequest.invoicePayableAt()))
+      .build();
 
-    Booking booking = bookingRepository.save(
```

Comment:
Was actually planning to check with you weather there was a neat way to generate the initial event from the statemachine instead of creating it here, and then went ahead and fixed it :)